extensions/objects/templates/literally-canvas-diagram-editor.component.ts

```diff
@@ -0,0 +1,102 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for literally canvas diagram editor.
+ */
+
+const LC = require('literallycanvas');
+require('literallycanvas/lib/css/literallycanvas.css');
+require('services/literally-canvas-helper.service.ts');
+
+angular.module('oppia').component('literallyCanvasDiagramEditor', {
+  template: require('./literally-canvas-diagram-editor.component.html'),
+  controller: [
+    'LiterallyCanvasHelperService', function(LiterallyCanvasHelperService) {
+      const ctrl = this;
+      var MAX_DIAGRAM_WIDTH = 491;
+      var MAX_DIAGRAM_HEIGHT = 551;
+      var DEFAULT_STROKE_WIDTH = 2;
+      var STROKE_WIDTH = [1, 2, 3, 5, 30];
+      ctrl.diagramWidth = 450;
+      ctrl.currentDiagramWidth = 450;
+      ctrl.diagramHeight = 350;
+      ctrl.currentDiagramHeight = 350;
+      ctrl.data = {};
+      ctrl.onWidthInputBlur = function() {
+        if (ctrl.diagramWidth < MAX_DIAGRAM_WIDTH) {
+          ctrl.currentDiagramWidth = ctrl.diagramWidth;
+          ctrl.lc.setImageSize(
+            ctrl.currentDiagramWidth, ctrl.currentDiagramHeight);
+        }
+      };
+
+      ctrl.onHeightInputBlur = function() {
+        if (ctrl.diagramHeight < MAX_DIAGRAM_HEIGHT) {
+          ctrl.currentDiagramHeight = ctrl.diagramHeight;
+          ctrl.lc.setImageSize(
+            ctrl.currentDiagramWidth, ctrl.currentDiagramHeight);
+        }
+      };
+
+      ctrl.getDiagramSizeInfo = function() {
+        var maxWidth = MAX_DIAGRAM_WIDTH;
+        var maxHeight = MAX_DIAGRAM_HEIGHT;
+        return 'This diagram has a maximum dimension of ' + maxWidth +
+        'px X ' + maxHeight + 'px to ensure that it fits in the card.';
+      };
+
+      ctrl.validate = function(data) {
+        // Will be implemented once data is saved.
+        return false;
+      };
+
+      ctrl.$onInit = function() {
+        LC.defineSVGRenderer(
+          'Rectangle', LiterallyCanvasHelperService.rectangleSVGRenderer);
+        LC.defineSVGRenderer(
+          'Ellipse', LiterallyCanvasHelperService.ellipseSVGRenderer);
+        LC.defineSVGRenderer(
+          'Line', LiterallyCanvasHelperService.lineSVGRenderer);
+        LC.defineSVGRenderer(
+          'LinePath', LiterallyCanvasHelperService.linepathSVGRenderer);
+        LC.defineSVGRenderer(
+          'Polygon', LiterallyCanvasHelperService.polygonSVGRenderer);
+        LC.defineSVGRenderer(
+          'Text', LiterallyCanvasHelperService.textSVGRenderer);
+        ctrl.lc = LC.init(document.getElementById('lc'), {
```

Comment:
I'm not sure this is OK. What prevents more than one element having the ID "lc", especially if this component appears in multiple RTEs that may coexist on the page at the same time?

You might need a different way of getting the specific element that corresponds to this instance of the component.

Reply:
Since literallyCanvas opens in a modal, there can't be multiple lc ids. But however I have solved the issue by using a random id that is unique to a particular component if we decide to use multiple literallyCanvas outside the modal.

Reply:
Done